### PR TITLE
Adapt webservice "getegrid" to new specification

### LINKED ISSUE #1187: By GNSS Coordinate (Type D)
__GetEGRID Service:__
---

By GNSS Coordinate (Type D)

**OLD:**

`https://example.com/oereb/getegrid/xml/{FLOAT,FLOAT}`

`https://example.com/oereb/getegrid/xml/46.94890,7.44665`

**NEW:**

`https://example.com/oereb/getegrid/xml?{GNSS=FLOAT,FLOAT}`

`https://example.com/oereb/getegrid/xml?GNSS=46.94890,7.44665`

Mandatory: GNSS

### LINKED ISSUE #1188: By Postal Adress (Type C)
__GetEGRID Service:__
---

By Postal Adress (Type C)

**OLD:**

`https://example.com/oereb/getegrid/xml/{INTEGER}/{STRING}/{STRING}`

`https://example.com/oereb/getegrid/xml/3084/Lindenweg/50a`

**NEW:**

`https://example.com/oereb/getegrid/xml?{POSTALCODE=INTEGER}&{LOCALISATION=STRING}&{NUMBER=STRING}`

`https://example.com/oereb/getegrid/xml?POSTALCODE=3084&LOCALISATION=Lindenweg&NUMBER=50a`

Mandatory: POSTALCODE, LOCALISATION, NUMBER

### LINKED ISSUE #1189: By XY (Type A)
__GetEGRID Service:__
---

By XY (Type A)

**OLD:**

LV95
`https://example.com/oereb/getegrid/xml/?{INTEGER,INTEGER}`

*OR*

LV03
`https://example.com/oereb/getegrid/xml/?{INTEGER,INTEGER}`

`https://example.com/oereb/getegrid/xml?XY=608000,228000`

**NEW:**

LV95
`https://example.com/oereb/getegrid/xml/?{XY=FLOAT,FLOAT}`

`https://example.com/oereb/getegrid/xml?XY=2608000.123,1228000.123`

Mandatory: XY

### LINKED ISSUE #1190: GetEGRID() Service:
GetEGRID() Service:
---
**NEW:**
for all, optional parameter GEOMETRY returning optional attribut LIMIT containing the feature geometries
Optional: GEOMETRY returning LIMIT

### LINKED ISSUE #1191: By NBIDENT/IDENTDN and NUMBER (Type B)
__GetEGRID Service:__
---

By NBIDENT/IDENTDN and NUMBER (Type B)

**OLD:**

`https://example.com/oereb/getegrid/xml/{STRING}/{STRING}`

`https://example.com/oereb/getegrid/xml/BE0200000332/100`

**NEW:**

`https://example.com/oereb/getegrid/xml?{IDENTDN=STRING}&{NUMBER=STRING}`

Mandatory: IDENTDN & NUMBER

### LINKED ISSUE #1192: mandatory attribut TYPE
**GetEGRID() Service:**
---
**NEW:**
mandatory attribut TYPE returning real estate type from given enumeration 
(todo: add link to federal xml here)

Mandatory: TYPE

### LINKED ISSUE #1193: Geometry handling in getEGRID Service
__GetEGRID Service:__

**OLD:**

`https://example.com/oereb/getegrid/xml/{GEOMETRY}?XY=2608000.230,1228000.23`

`https://example.com/oereb/getegrid/xml/GEOMETRY?XY=2608000.230,1228000.23`

**NEW:**

`https://example.com/oereb/getegrid/xml/?{GEOMETRY=true}&XY=2608000.230,1228000.23`

*OR*

`https://example.com/oereb/getegrid/xml/?{GEOMETRY=false}&XY=2608000.230,1228000.23`

*OR*

`https://example.com/oereb/getegrid/xml/?XY=2608000.230,1228000.23`

### PR DESCRIPTION
**Pull request by kdeininger**
Closes #1187
Closes #1188
Closes #1189
Closes #1190
Closes #1191
Closes #1192
Closes #1193

### CODE REVIEW

**Review by marionb — COMMENTED** [review]
Do we have some example parameters to test this?
(I had a try with the new url structure https://example.com/oereb/getegrid/xml?{POSTALCODE=INTEGER}&{LOCALISATION=STRING}&{NUMBER=STRING} but think I did not find the good parameters to see a result.)

File: pyramid_oereb/lib/renderer/getegrid/xml_.py [hunk 1]
```diff
@@ -45,9 +55,22 @@ def __call__(self, value, system):
         template = templates.get_template('getegrid.xml')
         try:
             content = template.render(**{
-                'data': value
+                'data': value[0],
+                'params': self._params_,
+                'get_gml_id': self._get_gml_id,
```
[the remark this answers is not in the document]
Comment (vvmruder) [reply pyramid_oereb/lib/renderer/getegrid/xml_.py:60]: I think we pass a reference to a method here. So its ok to have it without the `()`

File: pyramid_oereb/views/webservice.py [hunk 2]
```diff
@@ -115,119 +101,145 @@ def get_capabilities(self):
...
-                geom_wkt = 'SRID={0};{1}'
-                if xy:
-                    geom_wkt = geom_wkt.format(Config.get('srid'),
-                                               self.__parse_xy__(xy, buffer_dist=1.0).wkt)
-                elif gnss:
-                    geom_wkt = geom_wkt.format(Config.get('srid'), self.__parse_gnss__(gnss).wkt)
-                processor = create_processor()
-                records = processor.real_estate_reader.read(params, **{'geometry': geom_wkt})
-                response = self.__get_egrid_response__(records)
+            output_format = self.__validate_format_param__(['xml', 'json'])
```
[the remark this answers is not in the document]
Comment (kdeininger) [reply pyramid_oereb/views/webservice.py:120]: It is used in different cases. For `versions`, `capabilites` and `getegrid` only XML and JSON are allowed, `extract` may also contain PDF and URL (new).

But I could add constants for these two options.

File: pyramid_oereb/lib/renderer/getegrid/xml_.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (marionb) [reply pyramid_oereb/lib/renderer/getegrid/xml_.py:60]: Sorry I missed that.

File: pyramid_oereb/views/webservice.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kdeininger) [reply pyramid_oereb/views/webservice.py]: > Do we have some example parameters to test this?
> (I had a try with the new url structure [https://example.com/oereb/getegrid/xml?{POSTALCODE=INTEGER}&{LOCALISATION=STRING}&{NUMBER=STRING}](https://example.com/oereb/getegrid/xml?%7BPOSTALCODE=INTEGER%7D&%7BLOCALISATION=STRING%7D&%7BNUMBER=STRING%7D) but think I did not find the good parameters to see a result.)

This one should work:
http://localhost:6543/oereb/getegrid/json/?POSTALCODE=4104&LOCALISATION=Grenzweg&NUMBER=1

**Review by vvmruder — APPROVED** [review]
Looks good and works as expected.
:+1:

**Pull request merged**